### ISSUE TITLE
[expo-updates] Crash at launch if updates configuration is not provided

### ISSUE THREAD

**Issue by sjchmiela** [expo/expo#10123]
### Summary of Issue 

I have built NCL on SDK39 on iOS Turtle staging and it crashed.

```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[5]'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23e3de6e __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff512a19b2 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff23ed09d1 _CFThrowFormattedException + 194
	3   CoreFoundation                      0x00007fff23edb426 -[__NSPlaceholderDictionary initWithObjects:forKeys:count:].cold.4 + 38
	4   CoreFoundation                      0x00007fff23e9c6e7 -[__NSPlaceholderDictionary initWithObjects:forKeys:count:] + 247
	5   CoreFoundation                      0x00007fff23e39d71 +[NSDictionary dictionaryWithObjects:forKeys:count:] + 49
	6   HostExpNclexp                       0x00000001054cc08e -[EXAppLoaderExpoUpdates _startLoaderTask] + 1064
	7   HostExpNclexp                  <…>
```

The stacktrace suggests `launchWaitMs` which indeed was true in my situation without `"updates"` key configured (see [NCL's `app.json`](https://github.com/expo/expo/blob/master/apps/native-component-list/app.json)).

https://github.com/expo/expo/blob/b2e3e13d20f5aa330bfcf7484df2eec5c664ab8c/ios/Exponent/Kernel/AppLoader/EXAppLoaderExpoUpdates.m#L286-L301

I will rebuild the app with `"updates"` provided and follow up with results.

**Comment by sjchmiela** [comment]
Aye, providing
```json
    "updates": {
      "checkAutomatically": "ON_LOAD",
      "fallbackToCacheTimeout": 0
    },
```
let the app "move further".

**Comment by esamelson** [comment]
sorry about that 😞 fix in #10129 !
